lib/backend/compat/compat.go

```diff
@@ -617,36 +664,134 @@ func toNodeDeleteComponents(d *model.KVPair) (primary *model.KVPair, optional []
 			Key: model.HostIPKey{nk.Hostname},
 		},
 		&model.KVPair{
-			Key: model.HostBGPConfigKey{
-				Hostname: nk.Hostname,
+			Key: model.NodeBGPConfigKey{
+				Nodename: nk.Hostname,
 				Name:     "ip_addr_v4",
 			},
 		},
 		&model.KVPair{
-			Key: model.HostBGPConfigKey{
-				Hostname: nk.Hostname,
+			Key: model.NodeBGPConfigKey{
+				Nodename: nk.Hostname,
 				Name:     "ip_addr_v6",
 			},
 		},
 		&model.KVPair{
-			Key: model.HostBGPConfigKey{
-				Hostname: nk.Hostname,
+			Key: model.NodeBGPConfigKey{
+				Nodename: nk.Hostname,
 				Name:     "as_num",
 			},
 		},
 		&model.KVPair{
-			Key: model.HostBGPConfigKey{
-				Hostname: nk.Hostname,
+			Key: model.NodeBGPConfigKey{
+				Nodename: nk.Hostname,
 				Name:     "network_v4",
 			},
 		},
 		&model.KVPair{
-			Key: model.HostBGPConfigKey{
-				Hostname: nk.Hostname,
+			Key: model.NodeBGPConfigKey{
+				Nodename: nk.Hostname,
 				Name:     "network_v6",
 			},
 		},
 	}
 
 	return primary, optional
 }
+
+// toDatastoreGlobalBGPConfigKey modifies the Global BGP Config key to the one required by
+// the datastore (for back-compatibility).
+func toDatastoreGlobalBGPConfigKey(key model.GlobalBGPConfigKey) model.GlobalBGPConfigKey {
+	switch key.Name {
+	case "AsNumber":
+		key = model.GlobalBGPConfigKey{Name: "as_num"}
+	case "LogLevel":
+		key = model.GlobalBGPConfigKey{Name: "loglevel"}
+	case "NodeMeshEnabled":
+		key = model.GlobalBGPConfigKey{Name: "node_mesh"}
+	}
+	return key
+}
+
+// toDatastoreGlobalBGPConfigList modifies the Global BGP Config List interface to the one required by
+// the datastore (for back-compatibility).
+func toDatastoreGlobalBGPConfigList(l model.GlobalBGPConfigListOptions) model.GlobalBGPConfigListOptions {
+	switch l.Name {
+	case "AsNumber":
+		l = model.GlobalBGPConfigListOptions{Name: "as_num"}
+	case "LogLevel":
+		l = model.GlobalBGPConfigListOptions{Name: "loglevel"}
+	case "NodeMeshEnabled":
+		l = model.GlobalBGPConfigListOptions{Name: "node_mesh"}
+	}
+	return l
+}
+
+// fromDatastoreGlobalBGPKey modifies the Global BGP Config key from the one required by
+// the datastore (for back-compatibility).
+func fromDatastoreGlobalBGPKey(key model.GlobalBGPConfigKey) model.GlobalBGPConfigKey {
+	switch key.Name {
+	case "as_num":
+		key = model.GlobalBGPConfigKey{Name: "AsNumber"}
+	case "loglevel":
+		key = model.GlobalBGPConfigKey{Name: "LogLevel"}
+	case "node_mesh":
+		key = model.GlobalBGPConfigKey{Name: "NodeMeshEnabled"}
+	}
+	return key
+}
+
+// toDatastoreGlobalBGPConfig modifies the Global BGP Config KVPair to the format required in the
+// datastore (for back-compatibility).
+func toDatastoreGlobalBGPConfig(d model.KVPair) *model.KVPair {
+	// Copy the KVPair, so we aren't modifying the original.
+	k := d.Key.(model.GlobalBGPConfigKey)
+	d.Key = toDatastoreGlobalBGPConfigKey(k)
```

Comment:
Hm, so we're copying it so that toDatastoreGlobalBGPConfigKey doesn't modify the original, but then we're overwriting the original in the same line?

Are we just keeping this so that we can use `k.Name` below? If so, why don't we either:
- Switch on the "datastore" representation of the name, OR
- Just save off the same and use that in the switch below.

Or, am I misunderstanding?